src/runtime/pythonengine.cs

```
@@ -567,6 +576,60 @@ public static void Exec(string code, IntPtr? globals = null, IntPtr? locals = nu
             }
         }
 
+        /// <summary>
+        /// Gets the native thread ID.
+        /// </summary>
+        /// <returns>The native thread ID.</returns>
+        public static ulong GetNativeThreadID()
+        {
+            if (Runtime.PyVersion >= new Version(3, 8))
+            {
+                dynamic threading = Py.Import("threading");
+                return threading.get_native_id();
+            }
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                return GetCurrentThreadId();
+            }
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                return (ulong)pthread_selfLinux();
+            }
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                return pthread_selfOSX();
+            }
+
+            return 0;
```

Should raise an exception if we can't find a suitable implementation.